packages/components/src/number-control/index.tsx

```diff
@@ -64,16 +65,22 @@ function UnforwardedNumberControl(
 	const mergedRef = useMergeRefs( [ inputRef, forwardedRef ] );
 
 	const isStepAny = step === 'any';
-	const baseStep = isStepAny ? 1 : ensureNumber( step );
+	let baseStep = ensureNumber( step );
+	let baseSpin = ensureNumber( spinFactor ) * baseStep;
+	if ( isStepAny ) {
+		baseStep = 1;
+		baseSpin = 1 * ensureNumber( spinFactor );
+	}
```

Comment:
```suggestion
	const baseStep = isStepAny ? 1 : ensureNumber( step );
	const baseSpin = ensureNumber( spinFactor ) * baseStep;
```

Nit: It seems like we can simplify and avoid the `let`s?